manifests/init.pp

```diff
@@ -40,18 +40,17 @@
   validate_re($cron_allow, '^(present)|(absent)$', "cron::cron_allow is ${cron_allow} and must be absent or present")
   validate_re($cron_deny, '^(present)|(absent)$', "cron::cron_deny is ${cron_deny} and must be absent or present")
 
-  case type($enable_cron) {
-    'string': {
-      validate_re($enable_cron, '^(true|false)$', "cron::enable_cron may be either 'true' or 'false' and is set to <${enable_cron}>")
-      $enable_cron_real = str2bool($enable_cron)
-    }
-    'boolean': {
-      $enable_cron_real = $enable_cron
-    }
-    default: {
-      fail('cron::enable_cron type must be true or false.')
-    }
+  if is_string($enable_cron) {
+    validate_re($enable_cron, '^(true|false)$', "cron::enable_cron may be either 'true' or 'false' and is set to <${enable_cron}>")
```

Comment:
This would remove the specific fail message. I am fine with this as the error messages from str2bool are good enough.

But I still prepfer this pattern:
<pre>
  if is_bool($enable_cron) == true {
    $enable_cron_real = enable_cron
  } else {
    $enable_cron_real = str2bool($enable_cron)
  }
</pre>
because if you trust str2bool() there is no need for validate_bool() here.